src/main/java/gregtech/client/model/modelfactories/ActiveVariantBlockBakedModel.java

```diff
@@ -1,126 +1,118 @@
 package gregtech.client.model.modelfactories;
 
+import gregtech.api.GTValues;
 import gregtech.api.block.VariantActiveBlock;
 import gregtech.client.utils.BloomEffectUtil;
-import gregtech.common.ConfigHolder;
-import net.minecraft.block.properties.IProperty;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.Minecraft;
 import net.minecraft.client.renderer.block.model.BakedQuad;
 import net.minecraft.client.renderer.block.model.IBakedModel;
 import net.minecraft.client.renderer.block.model.ItemOverrideList;
 import net.minecraft.client.renderer.block.model.ModelResourceLocation;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.util.BlockRenderLayer;
 import net.minecraft.util.EnumFacing;
 import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.ForgeHooksClient;
 import net.minecraftforge.client.MinecraftForgeClient;
+import net.minecraftforge.client.event.ModelBakeEvent;
 import net.minecraftforge.common.property.IExtendedBlockState;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.relauncher.Side;
 
 import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.List;
+import java.util.*;
+import java.util.function.BooleanSupplier;
 import java.util.stream.Collectors;
 
-import static gregtech.common.blocks.MetaBlocks.statePropertiesToString;
-
+@Mod.EventBusSubscriber(modid = GTValues.MODID, value = Side.CLIENT)
 public class ActiveVariantBlockBakedModel implements IBakedModel {
-    private final ThreadLocal<TextureAtlasSprite> particle;
 
-    ActiveVariantBlockBakedModel() {
-        this.particle = ThreadLocal.withInitial(() -> Minecraft.getMinecraft().getTextureMapBlocks().getMissingSprite());
+    private static final Map<ModelResourceLocation, ActiveVariantBlockBakedModel> INSTANCES = new Object2ObjectOpenHashMap<>();
+
+    private final ModelResourceLocation inactiveModelLocation;
+    private final ModelResourceLocation activeModelLocation;
+    @Nullable
+    private final BooleanSupplier bloomConfig;
+
+    private final ModelResourceLocation modelLocation;
+
+    public ActiveVariantBlockBakedModel(ModelResourceLocation inactiveModelLocation, ModelResourceLocation activeModelLocation, @Nullable BooleanSupplier bloomConfig) {
+        this.inactiveModelLocation = inactiveModelLocation;
+        this.activeModelLocation = activeModelLocation;
+        this.bloomConfig = bloomConfig;
+        this.modelLocation = new ModelResourceLocation(
+                new ResourceLocation(GTValues.MODID, "active_variant_block_" + inactiveModelLocation.getNamespace() + "_" + inactiveModelLocation.getPath()),
```

Comment:
Should the constructor for this baked model take a mod id, so as not to require models to be located in only the GT folder?

Reply:
`modelLocation` field is not for reading models from resources. It's only used as an internal ID to register the model, and has no correlation to model files. Gregtech's mod ID was used because (1) I figured the system is inherently from Gregtech, rather than the mod model was belong to, and (2) wanted to prevent scenario where the ID collides with another model added by the mod.